File: backend/src/main/java/com/flowlink/execution/dto/ExecutionDetail.java
package com.flowlink.execution.dto;

import com.flowlink.core.domain.ExecutionStatus;
import com.flowlink.core.domain.TriggerType;

import java.time.Instant;
import java.util.List;
import java.util.UUID;

public record ExecutionDetail(
        UUID id,
        UUID flowId,
        UUID flowVersionId,
        ExecutionStatus status,
        TriggerType trigger,
        String triggeredBy,
        Instant startedAt,
        Instant finishedAt,
        String error,
        List<NodeExecutionView> nodes,
        // client(클라이언트→서버) 모드 노드에서 중단된 경우, 브라우저가 대신 호출할 요청. 아니면 null.
        PendingClientRequest pendingClient,
        // 폼 전송(FORM) 노드에서 중단된 경우, 브라우저가 팝업으로 submit 할 폼 명세. 아니면 null.
        PendingFormRequest pendingForm,
        // 사용자 입력 대기(INPUT) 노드에서 중단된 경우, 브라우저가 띄울 입력 창 명세. 아니면 null.
        PendingInputRequest pendingInput,
        // 콜백 대기(WAIT) 노드에서 중단된 경우의 대기 정보(수신 URL·타임아웃). 아니면 null.
        PendingWaitRequest pendingWait
) {
}

File: backend/src/main/kotlin/com/flowlink/execution/dto/ExecutionDetail.kt
package com.flowlink.execution.dto

import com.flowlink.core.domain.ExecutionStatus
import com.flowlink.core.domain.TriggerType
import java.time.Instant
import java.util.UUID

data class ExecutionDetail(
    val id: UUID,
    val flowId: UUID,
    val flowVersionId: UUID,
    val status: ExecutionStatus,
    val trigger: TriggerType,
    val triggeredBy: String?,
    val startedAt: Instant,
    val finishedAt: Instant?,
    val error: String?,
    val nodes: List<NodeExecutionView>,
    // client(클라이언트→서버) 모드 노드에서 중단된 경우, 브라우저가 대신 호출할 요청. 아니면 null.
    val pendingClient: PendingClientRequest?,
    // form(폼 전송·팝업) 노드에서 중단된 경우, 브라우저가 팝업으로 제출할 폼 명세. 아니면 null.
    val pendingForm: PendingFormRequest?,
    // wait(콜백 대기) 노드에서 중단된 경우, 브라우저가 대기할 콜백 명세. 아니면 null.
    val pendingWait: PendingWaitRequest?,
    // input(사용자 입력) 노드에서 중단된 경우, 브라우저가 모달로 띄울 입력 명세. 아니면 null.
    val pendingInput: PendingInputRequest?
)
